### DISCUSSION TITLE
Any more info/tutorial on how to use different paths/layers of svg for multiple lights?

### DISCUSSION THREAD

**Discussion by kerhbal** [ExperienceLovelace/ha-floorplan#177] [Q&A] [answered]
Hi there, so I've followed the tutorial and made the examples work.
Although I found that in the examples, the element is either 
- another small svg(light in home example)
- or a path of a different layer(floor planner home example)
and for this particular tutorial, the layer is just filled with some dark color, and we change the opacity with css and rules. There is only one image in the svg.

I'm trying to find a solution that can involve 2 images so that the dark/light effect is more realistic.

So I'd like to know how to use an svg with 2 layers with 1 image in each of them, and adjust the opacity for a part in one layer so that the part of image in that layer is visible or not?
The structure I tried is below:
- layer 1
  - dark image
  - path1
  - path2
  - ...
- layer 2
  - light image

so far no good, I cannot change the visibility of that part of image just by simply change opacity of that path.

Hope anyone is trying the same thing or has done something similar. Thanks!

**Comment by OzGav** [comment, marked as answer]
Not sure if I understand the issue. The floor planner example shows how to alter opacity as you have noted. If you have your base layer as dark (for example) then you should find you can overlay a lighter image and adjust the opacity of that image with opacity. If you are trying to fade out one image whilst simultaneously fading in another then that is a different issue but in my experience not required.

Maybe this discussion will help as well particularly the CSS the OP uses for light-on and -off https://github.com/ExperienceLovelace/ha-floorplan/discussions/154#discussioncomment-1407930

**Reply by kerhbal** [reply]
@OzGav thanks for the reply!
And the discussion you mentioned is a great reference to what I want to do. Though I see there were multiple layers, and each layer I assume only contains one part of the area and all other parts are transparent. In this way they can be layered together.



So what I meant was that I'm looking for another solution rather than this one, in that solution I will only have 2 layers, each layer contains one image. and I'm wondering if there is a way I can adjust the opacity of only a part of upper layer. In the tutorial this can be done since the upper layer does not contain any image, I've tried to use the same way to apply to an upper layer that with image, but failed to do so.
To summarize it:
What I can do:
- change opacity of path when upper layer does not contain image
- change opacity of whole upper layer if it contains image

What I want to do:
- change opacity of a part of upper layer when it contains image.

Hope this is more clear. thanks

**Reply by OzGav** [reply]
Yes absolutely you should not think of affecting entire layers at all. Take any individual element (a PNG that you have inserted as your lights on image for example) and give that an ID the same as your HA light entity and it will appear and disappear depending on the light’s state.

**Reply by OzGav** [reply]
Also I have tens of elements per layer in my SVG of six layers. So where you wrote “  I assume only contains one part of the area and all other parts are transparent.” that is not necessary. You just have to make sure whatever image you want on top is higher up the list of objects

**Reply by kerhbal** [reply]
so the problem with multiple small pngs as you suggest is that each png is a rectangle: the actual content can be polygon for sure, BUT the png itself is a rectangle. I've tried to copy the weird selected shape from GIMP, and the result is always a rectangle in Inkscape. The content looks perfect, because the copied part includes the weird shape and surround it with transparent and form into a rectangle.
so the area corresponded to user interaction is always a rectangle, which does not align with actual room shape(like rooms are always with some angle based on view point). 
And the solution I'm looking for is somehow to get it done with a path that can be aggregated with an image. a path can solve the polygon shape here, but it cannot be filled with image content, what I can do is just like the tutorial, fill the path with some dark color, and change the opacity of that.

**Reply by OzGav** [reply]
Yes true. What I do in this situation is I have an invisible touch point which is whatever shape I like in the top layer which toggles a boolean_input and then I use an automation which keeps the state of the light in sync with the boolean. In this way the light image comes on and off regardless of how the light is actually switched on and off but I also have a touch point in whatever shape I like in the floorplan.

**Reply by kerhbal** [reply]
Great idea! I also found a solution, clip-path. basically I can clip the image by a path, so that the image interacts with user can have a weird shape! 
And I think your way requires more steps but is more powerful. since your image and touch point is decoupled, I assume you can show a bigger image with smaller touch point, in such way you will see some scattered light in living room that is coming from kitchen lamp, and while you are touching the living room, it will not accidentally trigger kitchen lamp.

**Reply by OzGav** [reply]
Yes the scattered light in other rooms is one reason I use this technique. The other reason I use it is I have icons outside my rooms to control fans, bedside lights and aircon. I can touch those circles and effects in the room will change.

**Reply by OzGav** [reply]
If you are happy your queried is answered it would be great if you could mark it as such. Happy New Year!

**Reply by kerhbal** [reply]
Oh yeah for sure. Thank you so much for this! I didn’t realize that the touch element and the element we change class setting can be different!

**Reply by kerhbal** [reply]
Happy new year!
